nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -229,60 +228,61 @@ def _get_subgraph_data_for_node(self, node: NNCFNode, nncf_graph: NNCFGraph) ->
         statistic_nodes = []
         subgraph_input_ids, subgraph_output_ids = [], []
 
-        def fill_subgraph_output_ids(edge: NNCFGraphEdge):
-            node = edge.to_node
-            if node in visited_nodes:
-                return
-            visited_nodes.append(node)
-
-            input_id = (node.node_name, edge.input_port_id)
-            output_id = (edge.from_node.node_name, edge.output_port_id)
-
-            # If we found a node with bias, we have to collect it as a statistic node,
-            # and its input for _collected_stat_inputs_map,
-            # which will be used during the collection of statistics for the next node.
-            if self._is_node_correctable(node, nncf_graph):
-                subgraph_output_ids.append(output_id)
-                self._collected_stat_inputs_map[input_id] = output_id
-
-                statistic_nodes.append(edge.from_node)
-                return
-
-            for output_edge in nncf_graph.get_output_edges(node):
-                fill_subgraph_output_ids(output_edge)
+        def fill_subgraph_output_ids(main_node: NNCFNode):
+            edges_queue = nncf_graph.get_output_edges(main_node)
+            while edges_queue:
+                edge = edges_queue.pop()
+                node = edge.to_node
+                if node in visited_nodes:
```

Comment:
Too many indents IMHO, I would put while body here and in the `fill_subgraph_input_ids` in a separate function

Reply:
Done.

Reply:
Nice!